sync3/request.go

```diff
@@ -426,7 +442,12 @@ func (r *Request) ApplyDelta(nextReq *Request) (result *Request, delta *RequestD
 	}
 	// new subscriptions are the delta between old room subs and the newly calculated ones
 	for roomID := range resultSubs {
-		if _, ok := r.RoomSubscriptions[roomID]; ok {
+		if oldSub, ok := r.RoomSubscriptions[roomID]; ok {
+			// if the subscription is different, mark it as a delta, else skip it as it hasn't changed
+			newSub := resultSubs[roomID]
+			if oldSub.RequiredStateChanged(newSub) || oldSub.TimelineLimit != newSub.TimelineLimit {
+				delta.Subs = append(delta.Subs, roomID)
+			}
```

Comment:
This is the bugfix-in-passing you mentioned, I think?

Reply:
Yes